src/Uno.UWP/Storage/Pickers/FileOpenPicker.iOS.cs

```diff
@@ -26,41 +27,62 @@ public partial class FileOpenPicker
-			}
-
-			var documentTypes = UTTypeMapper.GetDocumentTypes(FileTypeFilter);
-			using var documentPicker = new UIDocumentPickerViewController(documentTypes, UIDocumentPickerMode.Open);
-			documentPicker.AllowsMultipleSelection = multiple;
-
-			var completionSource = new TaskCompletionSource<NSUrl?[]>();
-
-			documentPicker.OverrideUserInterfaceStyle = CoreApplication.RequestedTheme == SystemTheme.Light ?
-				UIUserInterfaceStyle.Light : UIUserInterfaceStyle.Dark;
+			UIViewController viewController = new UIViewController();
 
-			documentPicker.ShouldShowFileExtensions = true;
-			documentPicker.Delegate = new FileOpenPickerDelegate(completionSource);
-
-			if (documentPicker.PresentationController != null)
+			try
 			{
-				documentPicker.PresentationController.Delegate = new FileOpenPickerPresentationControllerDelegate(completionSource);
-			}
-
-			await rootController.PresentViewControllerAsync(documentPicker, true);
+				var rootController = UIApplication.SharedApplication?.KeyWindow?.RootViewController;
+				if (rootController == null)
+				{
+					throw new InvalidOperationException("Root controller not initialized yet. FolderPicker invoked too early.");
+				}
+
+				var documentTypes = UTTypeMapper.GetDocumentTypes(FileTypeFilter);
+
+				switch (SuggestedStartLocation)
+				{
+					case PickerLocationId.PicturesLibrary:
+						viewController = new UIImagePickerController();
+						break;
+					default:
+						viewController = new UIDocumentPickerViewController(documentTypes, UIDocumentPickerMode.Open);
+						break;
+				}
+				var completionSource = new TaskCompletionSource<NSUrl?[]>();
+
+				if (viewController is UIDocumentPickerViewController documentPicker)
+				{
+					documentPicker.AllowsMultipleSelection = multiple;
+					documentPicker.ShouldShowFileExtensions = true;
+					documentPicker.Delegate = new FileOpenPickerDelegate(completionSource);
+				}
+
+				viewController.OverrideUserInterfaceStyle = CoreApplication.RequestedTheme == SystemTheme.Light ?
+					UIUserInterfaceStyle.Light : UIUserInterfaceStyle.Dark;
+
+				if (viewController.PresentationController != null)
+				{
+					viewController.PresentationController.Delegate = new FileOpenPickerPresentationControllerDelegate(completionSource);
+				}
+
+				await rootController.PresentViewControllerAsync(viewController, true);
+
+				var nsUrls = await completionSource.Task;
+				if (nsUrls == null || nsUrls.Length == 0)
+				{
+					return FilePickerSelectedFilesArray.Empty;
+				}
+
+				var files = nsUrls
+					.Where(url => url != null)
+					.Select(nsUrl => StorageFile.GetFromSecurityScopedUrl(nsUrl!, null)).ToArray();
+				return new FilePickerSelectedFilesArray(files);
 
-			var nsUrls = await completionSource.Task;
-			if (nsUrls == null || nsUrls.Length == 0)
+			}
+			finally
 			{
-				return FilePickerSelectedFilesArray.Empty;
+				viewController.Dispose();
```

Comment:
Why is the dispose needed? Is it what closes the controller somehow?

Reply:
It was to respect the using for the viewController. Not revelant anymore